### PR TITLE
[User Schedule] Add inline_most and inline_at

### PR DESCRIPTION
**Pull request by rdspring1**
This PR implements the `inline_most` and `inline_at` functions.

`inline_most` inlines the selected tensors to the right most allowed position.
`inline_at` inlines the selected tensors to an iterDomain position in the reference tensor.

### PR List:
1. #2282 
2. #2283 
3. #2284
4. #2285
5. #2286   **<<<  YOU ARE HERE**
6. #2287
7. #2288

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -3090,6 +3091,75 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+        }
+      },
+      py::arg("selected_tensors") = std::vector<Tensor>());
+  nvf_sched.def(
+      "inline_at",
+      [](FusionDefinition::SchedOperators& self,
+         Tensor tensor,
+         int64_t pos,
+         bool best_effort,
+         const std::vector<Tensor>& selected_tensors) {
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:3129]: ditto. regarding the selected_tensors, excluded_tensors.

File: tests/python/test_schedule_ops.py [hunk 2]
```diff
@@ -355,6 +355,93 @@ def schedule(self):
...
+                fd.sched.merge(self.t2, dim=0)
+                fd.sched.split(self.t2, dim=0, factor=128)
+                fd.sched.split(self.t2, dim=0, factor=4)
+                fd.sched.transform_like(self.t2)
+
+                fd.sched.parallelize(self.t2, axis := 0, ParallelType.grid_x)
+                fd.sched.parallelize(self.t2, axis := -1, ParallelType.block_x)
+                fd.sched.parallelize_like(self.t2)
+
+                fd.sched.inline_at(self.t2, pos=1)
```
Comment (jjsjann123) [inline tests/python/test_schedule_ops.py:438]: I'm a little uncomfortable about our code path coverage in tests.

File: csrc/python_frontend/python_bindings.cpp [hunk 3]
```diff
@@ -3090,6 +3091,75 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("selected_tensors") = std::vector<Tensor>(),
       py::arg("selected_parallel_types") = std::unordered_set<ParallelType>(),
       py::arg("propagate_padding") = true);
+  nvf_sched.def(
+      "inline_most",
+      [](FusionDefinition::SchedOperators& self,
+         const std::vector<Tensor>& selected_tensors) {
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp:3097]: In python, we don't have operator support to create a set of `Tensor` objects. e.g., hash function and equality operator

File: tests/python/test_schedule_ops.py (hunk 2 above)
Comment (rdspring1) [reply tests/python/test_schedule_ops.py:550]: I added a test case for the `selected_tensor` variant of `inline_most` and `inline_at`

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjsjann123) [reply csrc/python_frontend/python_bindings.cpp:3086]: Good point. Thought that's a solvable problem. :)
Not saying that we need to fix it here, I feel that's a better / more consistent signature though.

[the remark this answers is not in the document]
Comment (jjsjann123) [reply csrc/python_frontend/python_bindings.cpp:3086]: good point. Though that's a solvable problem.

We don't have to do it here, but might be a good clean up task to have a more consistent API.

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Pull request merged**